src/util.cpp

```diff
@@ -304,6 +304,7 @@ class ArgsManagerHelper {
     static inline bool GetNetBoolArg(const ArgsManager &am, const std::string& net_arg)
     {
         std::pair<bool,std::string> found_result(false,std::string());
+        LOCK(am.cs_args);
```

Comment:
I'd prefer to annotate this function and then take the lock where `GetNetBoolArg` is called (in `GetChainName`)